Don't create files in the source tree while building alaCarte

Scheirle opened the pull request:
Dynamic files in the source tree prevent the developer to have multiple build version (debug, release) at the same time.
This is especially annoying if you want to run the unit tests in debug and release mode, since this required you to manually clean the source tree.

Changes:
- Create the doxygen doc in the build directory
- Moved the default config (data/config/alacarte-maps.conf.in) in the build directory 
- Unit tests now use the build directory for files they create.
  - deleted `tests/data/output/`
  - moved `tests/data/input/` to `tests/data/`

@florianjacob please have a look at the `.travis-publish-docs.sh` changes

florianjacob commented:
I added a minor fix to .travis-publish-docs.sh. In general, this is a really nice fix. :sparkles: We should have done a long time ago. What we did instead is fight the symptoms by putting that changing stuff into .gitignore. :laughing:

As you have the required overview of what exactly has moved or is still generated somewhere, could you take a look at the .gitignore file?

I spotted these candidates that I guess can be removed with your change:
* data/config/alacarte-maps.conf
* doc/doxygen/

But these might not be the only ones. With that included I'd try to merge it and test whether docs publishing still works.

Scheirle commented:
Updated the `.gitignore` file.
I also removed some entries which should be ignored by the users and not the projects ignore file.
- KDevelop Project files
- `.directory`

florianjacob commented:
Great! :100:  Let's test then…
